includes/wp-admin/import/class-mastodon.php

```diff
@@ -0,0 +1,242 @@
+
+		if ( isset( $file['error'] ) ) {
+			echo '<p><strong>' . \esc_html( $error_message ) . '</strong><br />';
+			echo \esc_html( $file['error'] ) . '</p>';
+			return null;
+		} elseif ( ! \file_exists( $file['file'] ) ) {
+			echo '<p><strong>' . \esc_html( $error_message ) . '</strong><br />';
+			/* translators: File path. */
+			\printf( \wp_kses_post( \__( 'The export file could not be found at <code>%s</code>. It is likely that this was caused by a permission problem.', 'activitypub' ) ), esc_html( $file['file'] ) );
+			echo '</p>';
+			return null;
+		}
+
+		\WP_Filesystem();
+
+		global $wp_filesystem;
+		$import_folder = $wp_filesystem->wp_content_dir() . 'import/';
+		$working_dir   = $import_folder . \basename( \basename( $file['file'], '.txt' ), '.zip' );
+
+		// Clean up working directory.
+		if ( $wp_filesystem->is_dir( $working_dir ) ) {
+			$wp_filesystem->delete( $working_dir, true );
+		}
+
+		// Unzip package to working directory.
+		\unzip_file( $file['file'], $working_dir );
+		$files = $wp_filesystem->dirlist( $working_dir );
+
+		if ( ! isset( $files['outbox.json'] ) ) {
+			echo '<p><strong>' . \esc_html( $error_message ) . '</strong><br />';
+			echo \esc_html__( 'The archive does not contain an Outbox file, please try again.', 'activitypub' ) . '</p>';
+		}
+
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		self::$outbox = json_decode( file_get_contents( $working_dir . '/outbox.json' ) );
+
+		$wp_filesystem->delete( $import_folder, true );
+
+		\wp_suspend_cache_invalidation();
+		\wp_defer_term_counting( true );
+		\wp_defer_comment_counting( true );
+
+		do_action( 'import_start' );
+
+		$result = self::import_posts();
+
+		\wp_suspend_cache_invalidation( false );
+		\wp_defer_term_counting( false );
+		\wp_defer_comment_counting( false );
+
+		\wp_import_cleanup( $file['id'] );
+
+		if ( is_wp_error( $result ) ) {
+			echo '<p><strong>' . \esc_html( $error_message ) . '</strong><br />';
+			echo esc_html( $result->get_error_message() ) . '</p>';
+		} else {
+			echo '<p>';
+			/* translators: Home URL */
+			\printf( \wp_kses_post( \__( 'All done. <a href="%s">Have fun!</a>', 'activitypub' ) ), esc_url( admin_url() ) );
+			echo '</p>';
+		}
+
+		do_action( 'import_end' );
+	}
+
+	/**
+	 * Process posts.
+	 *
+	 * @return true|\WP_Error True on success, WP_Error on failure.
+	 */
+	public static function import_posts() {
+		$user_id = 0;
+
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		foreach ( self::$outbox->orderedItems as $post ) {
+			// Skip boosts.
+			if ( 'Announce' === $post->type ) {
+				continue;
+			}
+
+			if ( ! $user_id ) {
+				$users = get_users(
+					array(
+						'fields'     => 'ID',
+						// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+						'meta_query' => array(
+							array(
+								'key'     => $GLOBALS['wpdb']->get_blog_prefix() . 'activitypub_also_known_as',
```

Comment:
So this will require this value to have been set first? 

And what about sites that only have the Blog User active?

Reply:
> So this will require this [also known as] value to have been set first?

Yes! We should probably add an author selection step, like the WP importer, to overwrite that, with the user we can find through this query as the default.

> And what about sites that only have the Blog User active?

Yeah, great question! Maybe another reason to add the author selection step?

Reply:
Author selection step sounds like a very good idea 👍🏻

Reply:
It exists now, if you want to give it another spin!